File: drawing-book/drawing-book.go
package drawingbook

import "math"

func pageCount(pageMax int32, pageTarget int32) int32 {
	fromStart := pageTarget - 1
	fromEnd := pageMax - pageTarget

	if fromStart < fromEnd {
		return int32(math.Ceil(float64(fromStart) / float64(2)))
	} else if (pageMax-pageTarget == 1) && (pageMax%2 == 0) {
		return (fromEnd / 2) + 1
	} else {
		return fromEnd / 2
	}
}

File: drawing-book/DrawingBook.java
public class DrawingBook {

  public static int pageCount(int pageMax, int pageTarget) {
    int fromStart = pageTarget - 1;
    int fromEnd = pageMax - pageTarget;

    if (fromStart < fromEnd) {
      return (int) Math.ceil(
        (double) fromStart / (double) 2
      );
    } else if (fromEnd == 1 && pageMax % 2 == 0) {
      return (fromEnd / 2) + 1;
    } else {
      return fromEnd / 2;
    }
  }

}
